miro_preprocessor/image_sorter/src/sorter_logic.py

```diff
@@ -6,23 +6,206 @@
+        return self._get('image_copyright_cleared') == 'Y'
+
+    @property
+    def is_cleared(self):
+        return self._get('image_cleared') == 'Y'
+
+    @property
+    def is_not_general_use(self):
+        return self._get('image_general_use') == 'N'
+
+    @property
+    def has_access_restrictions(self):
+        return not (self._is_blank("image_access_restrictions") or self._key_matches("image_access_restrictions",
+                                                                                     self._cc_accesses))
+
+    @property
+    def has_not_use_restrictions(self):
+        return not self._is_blank("image_use_restrictions") and self._key_matches("image_use_restrictions", self._cc_accesses)
+
+    @property
+    def is_not_for_public_access(self):
+        return (not self.is_copyright_cleared) or self.is_not_general_use or self.has_access_restrictions
+
+    @property
+    def is_for_public_access(self):
+        return not self.is_not_for_public_access
+
+    @property
+    def satisfies_api_filters(self):
+        return self.is_copyright_cleared and self.is_cleared and self.has_not_use_restrictions
+
+    @property
+    def is_cold_store(self):
+        return self.is_collection("F") or \
+            (self.is_collection("L", "M", "V") and self.image_library_dept_is_Archives_and_Manuscripts) or \
+            (self.is_collection("L", "M", "V") and self.image_tech_captured_mode_is_videodisc) or \
+            (self.is_collection("L", "M", "V") and
+             not self.is_innopac_id_8_digits and
+             self.is_title_empty and
+             self.is_image_pub_title_blank and
+             self.is_image_pub_periodical_blank)
+
+    @property
+    def is_tandem_vault(self):
+        return self.image_library_dept_is_Public_programmes or \
+            self.is_collection("L") and self.is_after_first_march_2016 or \
+            self.is_collection("L", "M", "V") and self.is_not_for_public_access
+
+    @property
+    def is_digital_library(self):
+        return not self.image_library_dept_is_Public_programmes \
+            and self.is_for_public_access and self.is_innopac_id_8_digits
+
+    @property
+    def is_catalogue_api(self):
+        return (not self.is_tandem_vault or self.is_digital_library) and self.satisfies_api_filters
 
 
 class Decision(enum.Enum):
     cold_store = 'cold_store'
     tandem_vault = 'tandem_vault'
+    digital_library = 'digital_library'
     catalogue_api = 'catalogue_api'
+    none = 'none'
 
 
-class Undecidable(Exception):
+class InvalidCollectionException(Exception):
     pass
 
 
 def sort_image(collection, image_data):
-    # For now reproduce the behaviour of the miro_adapter and send all images to the catalogue API.
-    return Decision.catalogue_api
+    print(collection)
+    print(image_data)
+    r = Rules(collection, image_data)
+
+    decisions = []
+
+    if not r.is_collection("F", "L", "V", "M"):
+        raise InvalidCollectionException
```

Comment:
Maybe include the metadata in the exception? Then you can search logs for the exception and it's all on one line, rather than searching and then having to work back up.

(And if we have Lambdas running concurrently, that may get harder)